### PR TITLE
feat: added ability to define the nodePort in the trino service

### PR DESCRIPTION
**Pull request by dantonbertuol**
Hello.

This PR is to be able to customize the NodePort in the Trino service.

I adjusted it due to the issue https://github.com/trinodb/charts/issues/203.

### CODE REVIEW

File: charts/trino/templates/service.yaml [hunk 1]
```diff
@@ -14,6 +14,9 @@ spec:
       targetPort: http
       protocol: TCP
       name: http
+      {{- if .Values.service.nodePort }}{{- if eq .Values.service.type "NodePort" }}
```
Comment (nineinchnick) [inline charts/trino/templates/service.yaml:17]: I don't think the condition on `.Values.service.type` is required. It's better to have the deployment fail with an actionable error from Kubernetes (that you can't set a `nodePort` for a non-node-port service type), than silently ignore a value that has been set explicitly. It makes it harder to debug, and we don't know what which service type should be used.

File: charts/trino/values.yaml [hunk 2]
```diff
@@ -348,6 +348,9 @@ service:
   annotations: {}
   type: ClusterIP
   port: 8080
+# -- The port the service listens on.
```
Comment (nineinchnick) [inline charts/trino/values.yaml:351]: If the property is commented out, it won't be added to the README. Can we have it without a default value (empty string)? Also, please document the default behavior - that the node port is selected by random, you can even reference the official Kubernetes documentation, using Markdown links: https://kubernetes.io/docs/concepts/services-networking/service/#nodeport-custom-port

Comment (nineinchnick) [reply charts/trino/values.yaml:351]: Or start the comment with a fully qualified property name, like `# service.nodePort -- The port the service listens on.`. I'm not sure if that'll work when the property is commented out, but let's see.

File: charts/trino/templates/service.yaml (hunk 1 above)
Comment (dantonbertuol) [reply charts/trino/templates/service.yaml]: Hi, I removed the "NodePort" type check. I also tested with the ClusterIP type and assigning a nodePort, and kubernetes returns the error:
`* Service "my-release-trino" is invalid: spec.ports[0].nodePort: Forbidden: may not be used when `type` is 'ClusterIP'`

File: charts/trino/values.yaml (hunk 2 above)
Comment (dantonbertuol) [reply charts/trino/values.yaml]: I standardized the comment by indicating `# service.nodePort` and also included the official documentation.

I also tested by inserting a string [""] and the behavior is null, that is, a port is randomly assigned.

**Comment by nineinchnick** [comment]
Please update the README - see the CI run failure: https://github.com/trinodb/charts/actions/runs/10386833434/job/28759113778?pr=204

Also, squash the commits into a single one. I can merge this as soon as the CLA clears out.

**Comment by dantonbertuol** [comment]
> Please update the README - see the CI run failure: https://github.com/trinodb/charts/actions/runs/10386833434/job/28759113778?pr=204
> 
> Also, squash the commits into a single one. I can merge this as soon as the CLA clears out.

I changed it as per your request.

**Comment by dantonbertuol** [comment]
@nineinchnick This error is a little strange, don't you think?
https://github.com/trinodb/charts/actions/runs/10387904546/job/28762488147?pr=204

**Comment by dantonbertuol** [comment]
> I copied those two lines into a text editor and only then noticed that there are two spaces missing at the end of the line. This is needed by markdown to correctly wrap the following lines into the bullet point.
> 
> I should have mentioned this earlier - you shouldn't edit the file manually, but use helm-docs to generate it automatically. There's a pre-commit hook that you could install to have this done automatically when you commit any other changes.

Ah, thanks for the information, I adjusted the readme again.

**Comment by dantonbertuol** [comment]
Hi @nineinchnick!

My CLA has been completed.

**Review by nineinchnick — APPROVED** [review]
